src/bridge/cockpitrouter.c

```diff
@@ -846,7 +847,12 @@ process_open (CockpitRouter *self,
         }
     }
   if (new_payload)
-    g_bytes_unref (new_payload);
```

Comment:
That's coming from cockpitjson, which also makes a billion other copies of it as it recursively traverses the data... I guess you know that, though.

In any case, I'd prefer to turn the logic inside out here: `GBytes` allows specifying a function to be called to free the data in question: we could use a function which clears the memory before it frees it.  Then we could just do a normal unref.  I'd find that a lot cleaner than the approach here (which might actually end up making an extra copy of the data, if a ref was held for some reason).

Reply:
Right -- we already do this in e.g. cockpitpeer.c:
```c

  self->authorize_values = g_hash_table_new_full (g_str_hash, g_str_equal,
                                                  g_free, clear_authorize_value);
```

And `clear_authorize_value()` is exactly such a wrapper which calls `cockpit_memory_clear()` and then `g_free()`.

More specifically with GBytes, this is already done in cockpitwebservice.c in process_socket_authorize():
```c
      bytes = g_bytes_new_with_free_func (password, strlen (password),
                                          clear_and_free_string, password);
```

Expanding that to cockpitjson requires some more thought, though: A lot of that goes through `cockpit_json_write_bytes()` which calls `g_bytes_new_take()` which has no "free func" argument. So we can't just apply this to `new_payload` straight away. We'll also need an API change to mark "secret" JSON to go via `cockpit_memory_clear()`, or just live with the (probably negligible?) overhead and just always clear it. But either way we need to make cockpitjson less copy trigger-happy, and work more with read-only references?